### PR TITLE
Add Firefox and Brave support for Trezor

### LINKED ISSUE #4904: Migrate to Trezor Connect v5
TREZOR  remade the whole library from scratch

https://github.com/trezor/connect/tree/v5

### PR DESCRIPTION
**Pull request by brunobar79**
This PR is blocked by https://github.com/MetaMask/eth-trezor-keyring/pull/11
Once it gets merged, I can point the eth-trezor-keyring to v0.2.0

This PR also adds errors messages returned by trezor and hides the export private key option for trezor accounts (since it's not supported)

Fixes #4904

### CODE REVIEW

File: app/manifest.json
Comment (brunobar79) [inline app/manifest.json]: I'm gonna try to take this permission of and see if stills works.
In theory if we're allowed to inject web3 on any website, we should be allowed to inject code on connect.trezor.io

Comment (brunobar79) [reply app/manifest.json]: YAY! we don't need it!

**Comment by danfinlay** [comment]
I tried this out:

When using the [personal_ecRecover](https://danfinlay.github.io/js-eth-personal-sign-examples/) method here, the signing request didn't always pop up. In fact, for a while, it wouldn't even add the signature request to the queue.

On my first signature (a personal_sign login to cryptokitties) After completing a signature, the window closed to reveal the MetaMask full screen page instead of taking me back to the page that requested the signature. I then tested a simple send and a contract publish, and both behaved appropriately.

I'd like to see if we can figure out why those are happening before merging.

Also: At what point can we host the trezor connect pages locally?

**Review by danfinlay — DISMISSED** [review]
Hoping we can get the tab management sorted out with Trezor first.

**Comment by brunobar79** [comment]
@danfinlay  I got this working with ganache-cli.  I was able to send to  and from the Trezor account connected to a local instance of ganache. The only consideration is that you need to initialize ganache with a custom network id  like this `ganache-cli -i 10`  where the value of i needs to be < 2^32 (based on [this answer](https://github.com/trezor/trezor-core/issues/346#issuecomment-418451665)).

Before merging this one in I need to merge MetaMask/eth-trezor-keyring#11 first, release to npm and update the version of eth-trezor-keyring here.

**Comment by kumavis** [comment]
@brunobar79 is this ready to be reviewed and merged?

**Comment by brunobar79** [comment]
@kumavis Almost...

Needs review and it's still blocked by https://github.com/MetaMask/eth-trezor-keyring/pull/11

**Comment by brunobar79** [comment]
Closing in favor of #5797

**Pull request closed without merging**
